src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,277 @@
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = data.query.$ascending || true;
+		const propertyFilter = data.query.propertyFilter || '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || ''; // will give pictures of flowers as default
+
+		// const filterOptions = data.query.filterOptions
+
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url: `${ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=${sortProperties}&sortProperties=cm%3Amodified&sortAscending=${sortAscending}&sortAscending=false&propertyFilter=${propertyFilter}&`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] },
+				],
+				facettes: ['cclom:general_keyword'],
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+
+		let eduResponse;
+		try {
+			eduResponse = await request(options);
+		} catch (e) {
+			// eslint-disable-next-line no-console
+			console.error('error: ', e);
+		}
+
+		const parsed = JSON.parse(eduResponse);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
+			mimetype: ['text/html', 'image/jpeg'],
+			provider: ['BauhausMaterial.de', 'München educationcenter', 'Khan Academy'],
+		};
+
+		// filter away everything buy selected mimetype
+		const filterMime = (obj, mimetypes) => {
+			let result = obj.nodes;
+			mimetypes.forEach((type) => { result = result.filter((n) => n.mimeType === type); });
+			return result;
+		};
+
+		// filter away everything buy selected providers
+		const filterProvider = (obj, providers) => obj === providers;
+
+		// adds accesstoken to image-url to let user see the picture on client-side.
+		if (parsed && parsed.nodes) {
+			parsed.nodes.forEach((node) => {
+				if (node.preview && node.preview.url) {
+					node.preview.url += `&accessToken=${this.accessToken}`;
+				}
+			});
+		}
+
+		const filterResult = (obj, opt) => {
+			let result;
+			// checks if user has set type filter
+			if (opt.mimetype.length) {
+				result = filterMime(obj, opt.mimetype);
+			}
+			// checks if user has set provider filter
+			if (opt.provider.length) {
+				result = filterProvider(obj, opt.provider);
+			}
+			return result;
+		};
+
+		// checks if user has set filter options
+		/* if (Object.values(filterOptions).length) {
+			parsed = filterResult(parsed, filterOptions);
+		} */
+		return parsed;
+	}
+
+	async GETONE(id, params) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['id' is defined but never used. Allowed unused args must match /app|req|res|next|options|params/u.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4977295)